src/main/java/lt/vu/trip/service/user/UserServiceImpl.java

```diff
@@ -1,13 +1,31 @@
 package lt.vu.trip.service.user;
 
 import lt.vu.trip.entity.User;
+import lt.vu.trip.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.security.core.context.SecurityContextHolder;
 import org.springframework.stereotype.Component;
 
+import java.util.List;
+
 @Component
 class UserServiceImpl implements UserService {
 
+	@Autowired
+	private UserRepository userRepository;
+
 	public User getCurrentUser() {
 		return (User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
 	}
+
+	public List<User> searchUsers(String query) {
+		String[] queryParts = query.trim().split(" ");
```

Comment:
Maybe the number of results should be limited for example to 10.

Reply:
Done. Added Top10 in repository method.